### ISSUE TITLE
On rotation of footprint in multiple of 45, the pads are missing.

### ISSUE THREAD

**Issue by imrishabh18** [tscircuit/footprinter#93]
![Image](https://github.com/user-attachments/assets/9283fe56-80dd-482c-bae6-19b4287c3d66)

```
export default () => {
  return (
    <board height="43.18mm" width="18.54mm">
      <chip
        manufacturerPartNumber="ATmega328P"
        name="U1"
        schX={0}
        schY={8}
        pcbX={0}
        pcbY={-8}
        footprint="qfp32"
        pcbRotation={45}
      />
    </board>
  )
}
```

**Comment by ShiboSoftwareDev** [comment]
![Image](https://github.com/user-attachments/assets/c99544c2-5a1f-4cce-8841-c83f94da0052)
actually only 90 degree rotation works

**Comment by imrishabh18** [comment]
Yeah I know 90 multiples are working fine. But I was seeing few pcb boards and they were placing the chips in 45deg. So, probably we need to support that as well. I think probably it will more work to rotate the pads I am guessing

**Comment by ShiboSoftwareDev** [comment]
@imrishabh18 can you delete this issue and open it in snippets? Abse found a fix but transferring issues doesn't transfer bounties so please reopen it there. fwiw I implemented rotated_rect in all relevant repos except the autorouter repo. Can you please investigate this issue as I couldn't find the problme? https://github.com/tscircuit/autorouting/issues/96

**Comment by imrishabh18** [comment]
If not here then, I am guessing this logic needs to be implemented in `PcbViewer` about rotation of footprint. Why in snippets though? 🤔

cc: @seveibar

**Comment by ShiboSoftwareDev** [comment]
@imrishabh18 the original problem is the pads not appearing which is fixed by updating pcb-viewer in snippets since I already implemented it in pcb-viewr, and he found another problem where the rotation direction is reversed which needs to be fixed in pcb-viewer. Sorry I didn't clarify earlier
